charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -2,19 +2,80 @@
 Return the name of the configMap holding the Super Agent's config. Defaults to release's fill name suffiexed with "-config"
 */ -}}
 {{- define "newrelic-super-agent.config.name" -}}
-{{- (include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" )) -}}
+{{- include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" ) -}}
 {{- end -}}
 
+
+
+{{- /*
+Return the agents part that should go in the super agent config. It is created from `.Values.config.subAgents`.
+*/ -}}
+{{- define "newrelic-super-agent.config.agents.yaml" -}}
+{{- if (.Values.config).subAgents -}}
+{{- $agents := dict -}}
+{{- range $subAgentName, $subAgentConfig := (.Values.config).subAgents -}}
+  {{- if not ($subAgentConfig).type -}}
+    {{- fail (printf "Agent %s does not have agent type" $subAgentName) -}}
+  {{- end -}}
+  {{- $_ := dict $subAgentName (dict "agent_type" $subAgentConfig.type "content" $subAgentConfig.content) | mustMerge $agents -}}
```

Comment:
it always confuses me when in the functions we do `| variable` 
isn't easier to read `$agents = mustMerge $agents dict ...`

Reply:
Fair. I changed all of them on [#d7c11af](https://github.com/newrelic/helm-charts/pull/1466/commits/d7c11af329768d346ab361f224a45d0e5ce1944c).